test/gtest/ucp/test_ucp_am.cc

```diff
@@ -842,6 +842,180 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
 
 
+class test_ucp_am_nbx_send_copy_header : public test_ucp_am_nbx {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_AM, 0, "");
+        add_variant_with_value(variants, UCP_FEATURE_AM, UCP_AM_SEND_FLAG_REPLY,
+                               "reply");
+    }
+
+protected:
+    void test_copy_header_on_pending(size_t header_size, size_t data_size,
+                                     bool rndv = false)
+    {
+        ucs_status_ptr_t rndv_pending_sptr = NULL;
+        unsigned flags                     = UCP_AM_SEND_FLAG_COPY_HEADER;
+        unsigned data_cb_flags             = 0;
+        size_t fillsq_msg_size             = data_size;
+        mem_buffer sbuf(data_size, tx_memtype());
+        ucs_status_ptr_t pending_sptr;
+
+        sbuf.pattern_fill(SEED);
+        m_hdr_copy.resize(header_size);
+        ucs::fill_random(m_hdr_copy);
+        ucp::data_type_desc_t sdt_desc(m_dt, sbuf.ptr(), data_size);
+        m_hdr       = m_hdr_copy;
+        req_counter = 0;
+        cb_counter  = 0;
+
+        /* 
+         * Filling the SQ is done by sending messages defined
+         * according to the test case (short, bcopy, zcopy), except for RNDV.
+         * 
+         * For RNDV we use predefined message length to fill the SQ so we will not
+         * get IN_PROGRESS request when calling fill_sq.
+         * 
+         * We cannot use predefined message length in all the tests because when
+         * testing bcopy multi fragments we actually testing two test cases:
+         * - When pending request is returned for the first fragment.
```

Comment:
- Get a pending request that did not yet send the first fragment
- Get a pending request that sent the first fragment, but not completed sending all fragments.

The exact scenario depends on transport-specific properties, such as tx queue length and fragment size.